### PR TITLE
Expose travel assignments for users

### LINKED ISSUE #3421: Expose travel assignments in user API

### PR DESCRIPTION
**Pull request by kberzinch**
Closes #3421

### CODE REVIEW

File: app/Models/User.php [hunk 1]
```diff
@@ -568,6 +568,7 @@ public function getRelationshipPermissionMap(): array
             'rsvps' => 'rsvps',
             'roles' => 'roles-and-permissions',
             'permissions' => 'roles-and-permissions',
+            'assignments.travel' => 'users',
```
Comment (kberzinch) [inline app/Models/User.php:571]: `AuthorizeInclude` assumes the permission starts with `read-`, and there's just one permission for travel called `manage-travel`. I think it's fine to expose TAs to `read-users` but I'm open to other opinions on that.

I also do want to enforce eager-loading here.

File: app/Http/Resources/User.php [hunk 2]
```diff
@@ -94,6 +95,11 @@ public function toArray($request): array
                     $this->resource->relationLoaded('roles') ? $this->getAllPermissions()->pluck('name') : [],
                 ]
             ),
+            'travel' => $this->when(
+                ($this->requestingSelf($request) || Auth::user()->can('manage-travel')) &&
```
Comment (evan10s) [inline app/Http/Resources/User.php:99]: It seems like authorize include wants `read-users` but this if statement is looking for `manage-travel`, am I missing something?

Comment (kberzinch) [reply app/Http/Resources/User.php:99]: I guess the end result is you need both permissions. `read-users` is required to load the permission in the controller, `manage-travel` is required for it to populate in the resource here.

It's a little convoluted but I don't super want to add a `read-travel` permission just for this.

Comment (evan10s) [reply app/Http/Resources/User.php:99]: this seems like jank that might end up confusing someone later on. I would lightly prefer (4/10) if we did something to make it more clear what's happening. AuthorizeIncludes is a bit hard to understand for newer devs or if you haven't touched the code in a while and forgot it exists

Comment (kberzinch) [reply app/Http/Resources/User.php:99]: Fair enough. Let me play around with it a bit.

File: app/Traits/AuthorizeInclude.php [hunk 3]
```diff
@@ -45,9 +45,9 @@ public function authorizeInclude(string $class, ?string $requestedInclude = null
             $permission = array_key_exists(
                 $include,
                 $relationPermMap
-            ) ? $relationPermMap[$include] : $this->camelToDashed($include);
+            ) ? $relationPermMap[$include] : 'read-'.$this->camelToDashed($include);
 
-            if (Auth::user()->cant('read-'.$permission)) {
+            if (Auth::user()->cant($permission)) {
                 Log::debug('User is missing permission: read-'.$permission);
```
Comment (evan10s) [inline app/Traits/AuthorizeInclude.php:51]: this doesn't seem to account for the new definition of `$permission` above

**Review by evan10s — APPROVED** [review]

File: app/Traits/AuthorizeInclude.php (hunk 3 above)
Comment (kberzinch) [reply app/Traits/AuthorizeInclude.php:51]: Oop good catch

**Pull request merged**